### PR TITLE
Measures Screen

### PR DESCRIPTION
**Pull request by Gotalicp**
Moved devider to ui common
Changed Measure fragment to compose

Old: 
Screenshot_2024-11-11-22-34-13-975_bg zahov app
New:
Screenshot_2024-11-11-19-37-07-056_bg zahov app

### CODE REVIEW

File: app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt [hunk 1]
```diff
@@ -1,25 +1,71 @@
...
+/**
+ * Enum class representing different types of body measurements.
+ * Each measurement type has an associated key that describes the measurement in a user-friendly format.
+ *
+ * @property key A descriptive name for the measurement type.
+ */
 enum class MeasurementType(val key: String) {
+    /** Represents the user's weight measurement. */
     Weight("Weight"),
+
```
Comment (trifonov-i) [inline app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt:12]: Again for strings use stringResources

File: app/src/main/java/bg/zahov/app/ui/measures/MeasuresScreen.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.app.ui.custom.CommonDivider
+
+@Composable
+fun MeasuresScreen(navigateInfo: (String) -> Unit, viewModel: MeasuresViewModel = viewModel()) {
+    MeasuresContent(
+        items = viewModel.getAllMeasures(),
```
Comment (trifonov-i) [inline app/src/main/java/bg/zahov/app/ui/measures/MeasuresScreen.kt:27]: I know it's just a list with enums but create a state for consistency and separation of responsibilities

**Review by trifonov-i — CHANGES_REQUESTED** [review]

File: app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt (hunk 1 above)
Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt:12]: dont quite understand what i should do. Should i change the string in the data class to the id of the stringResources or ?

Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt:15]: change it

**Review by trifonov-i — APPROVED** [review]

File: app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt [hunk 3]
```diff
@@ -1,25 +1,71 @@
...
+     * Companion object for the [MeasurementType] enum, providing utility functions.
+     */
     companion object {
+        /**
+         * Finds the corresponding [MeasurementType] for a given key, ignoring case sensitivity.
+         *
+         * @param key The key to match against the [MeasurementType] enum.
+         * @return The matching [MeasurementType] if found, or `null` otherwise.
+         */
         fun fromKey(key: String): MeasurementType? =
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt:68]: As discussed in previous PRs this function is not needed. Use `entries..` instead wherever needed.

File: app/src/main/java/bg/zahov/app/ui/custom/CommonDivider.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import androidx.compose.ui.unit.dp
+import bg.zahov.fitness.app.R
+
+@Composable
+fun CommonDivider(modifier: Modifier = Modifier) = HorizontalDivider(
+    modifier = modifier
+        .fillMaxWidth()
+        .height(1.dp)
+        .padding(top = 4.dp, bottom = 16.dp),
+    color = colorResource(R.color.white)
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/ui/custom/CommonDivider.kt:19]: Use the theme colors instead.

File: app/src/main/java/bg/zahov/app/ui/measures/MeasuresFragment.kt [hunk 5]
```diff
@@ -49,23 +35,16 @@ class MeasuresFragment : Fragment() {
...
-                                measurementViewModel.onMeasurementClick(item.title)
-                                findNavController().navigate(
-                                    R.id.measures_to_measurement_info,
-                                    bundleOf(MEASUREMENT_ARGS to item.title)
-                                )
-                            }
-                        }
-                    updateItems(enumValues<MeasurementType>().map { enum -> AdapterMeasurement(enum.key) })
-                }
+        requireActivity().showTopBar()
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/ui/measures/MeasuresFragment.kt:38]:
```suggestion
        activity?.showTopBar()
```

File: app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt (hunk 3 above)
Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/data/model/MeasurementType.kt:68]: I know but if i change it i have to change a lot of files sometimes. Will change it in another pr

File: app/src/main/java/bg/zahov/app/ui/custom/CommonDivider.kt (hunk 4 above)
Comment (Gotalicp) [reply app/src/main/java/bg/zahov/app/ui/custom/CommonDivider.kt:19]: Will change it once we have a theme

**Pull request merged**